railgun/hack/generators/controllers/networking/main.go

```diff
@@ -20,103 +20,114 @@ const outputFile = "controllers/configuration/zz_generated_controllers.go"
+		Package:                  "k8s.io/api/core/v1",
+		Type:                     "Endpoints",
+		Plural:                   "endpoints",
+		URL:                      "v1",
+		CacheType:                "Endpoint",
+		RequiresIngressClassName: false,
 	},
 	typeNeeded{
-		PackageImportAlias: "netv1",
-		PackageAlias:       "NetV1",
-		Package:            "k8s.io/api/networking/v1",
-		Type:               "Ingress",
-		Plural:             "ingresses",
-		URL:                "networking.k8s.io",
-		CacheType:          "IngressV1",
+		PackageImportAlias:       "netv1",
+		PackageAlias:             "NetV1",
+		Package:                  "k8s.io/api/networking/v1",
+		Type:                     "Ingress",
+		Plural:                   "ingresses",
+		URL:                      "networking.k8s.io",
+		CacheType:                "IngressV1",
+		RequiresIngressClassName: true,
 	},
 	typeNeeded{
-		PackageImportAlias: "netv1beta1",
-		PackageAlias:       "NetV1Beta1",
-		Package:            "k8s.io/api/networking/v1beta1",
-		Type:               "Ingress",
-		Plural:             "ingresses",
-		URL:                "networking.k8s.io",
-		CacheType:          "IngressV1beta1",
+		PackageImportAlias:       "netv1beta1",
+		PackageAlias:             "NetV1Beta1",
+		Package:                  "k8s.io/api/networking/v1beta1",
+		Type:                     "Ingress",
+		Plural:                   "ingresses",
+		URL:                      "networking.k8s.io",
+		CacheType:                "IngressV1beta1",
+		RequiresIngressClassName: true,
 	},
 	typeNeeded{
-		PackageImportAlias: "extv1beta1",
-		PackageAlias:       "ExtV1Beta1",
-		Package:            "k8s.io/api/extensions/v1beta1",
-		Type:               "Ingress",
-		Plural:             "ingresses",
-		URL:                "apiextensions.k8s.io",
-		CacheType:          "IngressV1beta1",
+		PackageImportAlias:       "extv1beta1",
+		PackageAlias:             "ExtV1Beta1",
+		Package:                  "k8s.io/api/extensions/v1beta1",
+		Type:                     "Ingress",
+		Plural:                   "ingresses",
+		URL:                      "apiextensions.k8s.io",
+		CacheType:                "IngressV1beta1",
+		RequiresIngressClassName: true,
 	},
 	typeNeeded{
-		PackageImportAlias: "kongv1",
-		PackageAlias:       "KongV1",
-		Package:            "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1",
-		Type:               "KongIngress",
-		Plural:             "kongingresses",
-		URL:                "configuration.konghq.com",
-		CacheType:          "KongIngress",
+		PackageImportAlias:       "kongv1",
+		PackageAlias:             "KongV1",
+		Package:                  "github.com/kong/kubernetes-ingress-controller/railgun/apis/configuration/v1",
+		Type:                     "KongIngress",
+		Plural:                   "kongingresses",
+		URL:                      "configuration.konghq.com",
+		CacheType:                "KongIngress",
+		RequiresIngressClassName: true,
```

Comment:
Despite the (evergreen source of confusion) name, KongIngress is not an Ingress-like, and is only loaded when some other resource requests it. It does not use class information itself.
```suggestion
		RequiresIngressClassName: false,
```